packages/orm/src/collections/ManyToOneReference.ts

```diff
@@ -1,14 +1,26 @@
 import { Entity, EntityConstructor, getMetadata, IdOf, isEntity } from "../EntityManager";
-import { ensureNotDeleted, fail, getEm, maybeResolveReferenceToId, Reference, setField } from "../index";
-import { OneToManyCollection } from "./OneToManyCollection";
+import {
+  ensureNotDeleted,
+  fail,
+  getEm,
+  maybeResolveReferenceToId,
+  OneToOneReference,
+  Reference,
+  setField,
+} from "../index";
 import { AbstractRelationImpl } from "./AbstractRelationImpl";
+import { OneToManyCollection } from "./OneToManyCollection";
 
 /**
  * Manages a foreign key from one entity to another, i.e. `Book.author --> Author`.
  *
  * We keep the current `author` / `author_id` value in the `__orm.data` hash, where the
  * current value could be either the (string) author id from the database, or an entity
  * `Author` that the user has set.
+ *
+ * Note that if our `images.author_id` column is unique, this `ManyToOneReference` will essentially
+ * be half of a one-to-one relationship, but we'll keep using this reference on the "owning"
+ * side; the other side, i.e. `Author.image` will use a `OneToOneReference` to point back to us.
```

Comment:
Was wondering how you were going to do this.  Is there a technical limitation to making it just o2o on both sides?  IMO that would be preferable.  In practice most people are just going to use `entity.relation.get[0]` anyways.  It'd be nice if we could make it both cleaner for the developer and the type system.  Also then all the code related to a o2o could be in o2o.  Seems like it could just generate a different query for load based on which end of the relationship it's on.

Reply:
> o2o on both sides

Well, it effectively looks like a o2o on both sides, i.e. both sides are `image.book = ...` and `book.image = ...`. This is just the impl caveat that the side that "owns the fk" still uses the ManyToOneReference class.

> In practice most people are just going to use entity.relation.get[0] anyways

Hm, not following?

> Seems like it could just generate a different query for load based 

Hm, potentially, but I think the differences of "one side is based on the column in the table" and "the other side is always a lookup" are big-ish. I.e. it shows up has the "column in table side" could be nullable/not-nullable (based on not null on the column) but the other is always nullable (b/c we can't enforce at the db level that the pk of the current row must exist on a row in the other table). 

I was kinda surprised at how little had to change in ManyToOne so will probably keep for now.

Does seem like a collection refactoring is coming up at some point.